File: 1_CvsPHP/src/2.pi.calculation.c
/* Homework C vs. PHP Performance Analysis 2.pi.calculation.c
 * 2017-2 Internet Programming
 * Written by Lee Tae Hee, Dept. of Computer Science and Engineering, Univ. of Seoul
 */

#include <stdio.h>
#include <time.h>

int main() {
	time_t start_time = 0, end_time = 0;
	double gap;
      int lmt = 0;
      scanf("%d",&lmt);
	start_time = clock();
	double pi = 0;
	int s = 1;
	for (int i = 1; i <= lmt; i++) {
		pi += 1 / (double)(2 * i - 1)*s;
		s *= -1;
	}
	pi *= 4;

	end_time = clock();
	gap = (double)(end_time - start_time) / (CLOCKS_PER_SEC);
	printf("result : %.20lf \n", pi);
	printf("%lf ÃÊ\n", gap);

	return 0;
}

File: 1_CvsPHP/src/2.pi.calculation.php
/* Homework C vs. PHP Performance Analysis 2.pi.calculation.php
 * 2017-2 Internet Programming
 * Written by Choe Hyeong Jin, Dept. of Computer Science and Engineering, Univ. of Seoul
 */

<!DOCTYPE html>
<meta charset="utf-8"/>

<?PHP
	function get_time(){
		list($usec, $sec) = explode(" ", microtime());
		return ((float)$usec + (float)$sec);
	}

	$LMT = 100000;

	$start = get_time();
	
	$k = 1;
	$result = 0;
	$s=-1;

	for($k = 2; $k <= $LMT; $k++){
		$result += 1/(2*$k-1)*$s;
		$s*=-1; 
	}
	$result = 4 * $result;

	$end = get_time();

	$time = $end - $start;
	echo 'Pi calculation using Gregory-Leibniz series...'.$LMT.' = '.$result.'
';
	echo 'It takes '.$time.' seconds...
';
?>
